cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -281,3 +289,53 @@ func GetKubeletDir(dryRun bool) (string, error) {
 	}
 	return kubeadmconstants.KubeletRunDirectory, nil
 }
+
+func UpdateKubeletLocalMode(cfg *kubeadmapi.InitConfiguration, dryRun bool) error {
+	// TODO(chrischdi): how to get the correct dir? kubeadm init has a flag to change the location
+	dir := kubeadmconstants.KubernetesDir
+
+	kubeletKubeConfigFilePath := filepath.Join(dir, kubeadmconstants.KubeletKubeConfigFileName)
+
+	if _, err := os.Stat(kubeletKubeConfigFilePath); err != nil {
+		if os.IsNotExist(err) {
+			// TODO(chrischdi): should we print a warning or even return the error?
+			return nil
+		}
```

Comment:
Should we ignore and skip if the file does not exist? or fail?

Reply:
we could have a warning.

klog.Warningf("could not mutate the Server field in %s: %v", err)
return nil